onnxruntime/contrib_ops/cpu/layer_norm.cc

```diff
@@ -12,25 +12,19 @@
 namespace onnxruntime {
 namespace contrib {
 
-#define REGISTER_KERNEL_TYPED(T)                                  \
-  ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
-      LayerNormalization,                                         \
-      kOnnxDomain,                                                \
-      1,                                                          \
-      T,                                                          \
-      kCpuExecutionProvider,                                      \
-      KernelDefBuilder()                                          \
-          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
-      LayerNorm<T, false>);                                       \
-  ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
-      SimplifiedLayerNormalization,                               \
-      kOnnxDomain,                                                \
-      1,                                                          \
-      T,                                                          \
-      kCpuExecutionProvider,                                      \
-      KernelDefBuilder()                                          \
-          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
-      LayerNorm<T, true>);
+#define REGISTER_KERNEL_TYPED(T)                                                                        \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(LayerNormalization, kOnnxDomain, 1, T, kCpuExecutionProvider,           \
+                                KernelDefBuilder()                                                      \
+                                    .TypeConstraint("T", DataTypeImpl::GetTensorType<T>())              \
+                                    .TypeConstraint("U", DataTypeImpl::GetTensorType<T>())              \
+                                    .TypeConstraint("V", DataTypeImpl::GetTensorType<T>()),             \
+                                LayerNorm<T, false>);                                                   \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(SimplifiedLayerNormalization, kOnnxDomain, 1, T, kCpuExecutionProvider, \
+                                KernelDefBuilder()                                                      \
+                                    .TypeConstraint("T", DataTypeImpl::GetTensorType<T>())              \
+                                    .TypeConstraint("U", DataTypeImpl::GetTensorType<T>())              \
+                                    .TypeConstraint("V", DataTypeImpl::GetTensorType<T>()),             \
+                                LayerNorm<T, true>);
 
```

Comment:
We need the hash to be stable for backwards compatibility. As we don't have a good way to handle completely new constraints being added (we can handle the types in the list changing) without an opset change (contrib ops are all opset 1), we have a list of hardcoded exceptions in UpdateHashForBackwardsCompatibility in \onnxruntime\core\framework\kernel_def_hash_helpers.cc. 

Please add a mapping from the original hash to the new one for the 4 CPU contrib ops there (and note exactly which registration, including type, each hash is for in the comment).

Reply:
Added. I cann't tell which ones are for float or double, so I assume the first ones in contrib.cpu.json is for float. And we don't need this for training ops, right?